examples/latent_sde.py

```diff
@@ -71,23 +71,47 @@ def f(self, t, y):  # Approximate posterior drift.
         inp = torch.cat((torch.sin(t), torch.cos(t), y), dim=-1)
         return self.net(inp)
 
+    def f_aug(self, t, y):
+        y = y[:, 0][:, None]
+        res = self.f(t, y)
+        z = res - self.h(t, y)
+        g = self.g(t, y)
+        g = torch.where(
+            g.abs() > 1e-7,
+            g,
+            torch.ones_like(g).fill_(1e-7) * g.sign()
+        )
+        z = z / g
+        z = .5 * (torch.norm(z, dim=1) ** 2.)[:, None]
+        return torch.cat([res, z], dim=1)
+
     def g(self, t, y):  # Shared diffusion.
         return self.sigma.repeat(y.size(0), 1)
 
+    def g_aug(self, t, y):
+        y = y[:, 0][:, None]
+        res = self.g(t, y)
+        z = torch.zeros_like(y)
+        return torch.cat([res, z], dim=1)
+
     def forward(self, ts, batch_size, eps=None):
-        eps = torch.randn(batch_size, 1).to(self.qy0_std) if eps is None else eps
+        eps = torch.cat([torch.randn(batch_size, 1).to(self.qy0_std), torch.zeros(batch_size, 1)
+                        .to(self.qy0_std)], dim=1) if eps is None else eps
         y0 = self.qy0_mean + eps * self.qy0_std
 
         qy0 = Normal(loc=self.qy0_mean, scale=self.qy0_std)
         py0 = Normal(loc=self.py0_mean, scale=self.py0_std)
         logqp0 = kl_divergence(qy0, py0).sum(1).mean(0)  # KL(time=0).
 
         if args.adjoint:
-            zs, logqp = sdeint_adjoint(self, y0, ts, logqp=True, method=args.method, dt=args.dt, adaptive=args.adaptive,
-                                       rtol=args.rtol, atol=args.atol)
+            zs = sdeint_adjoint(self, y0, ts, method=args.method, dt=args.dt, adaptive=args.adaptive,
+                                       rtol=args.rtol, atol=args.atol, names={'drift': 'f_aug', 'diffusion': 'g_aug'})
         else:
-            zs, logqp = sdeint(self, y0, ts, logqp=True, method=args.method, dt=args.dt, adaptive=args.adaptive,
-                               rtol=args.rtol, atol=args.atol)
+            zs = sdeint(self, y0, ts, method=args.method, dt=args.dt, adaptive=args.adaptive,
+                               rtol=args.rtol, atol=args.atol, names={'drift': 'f_aug', 'diffusion': 'g_aug'})
+        logqp = zs[:, :, 1]
+        zs = zs[:, :, 0][:, :, None]
+
         logqp = logqp.sum(0).mean(0)
```

Comment:
Now we should only take the last entry of `logqp`.

Reply:
Done!